test/governon.flow.test.ts

```diff
@@ -0,0 +1,374 @@
+
+    await token2.connect(owner).mint(voter2.address, "9");
+
+    await token.connect(proposer).delegate(proposer.address);
+    await token.connect(voter1).delegate(voter1.address);
+    await token.connect(voter2).delegate(voter2.address);
+
+    const proposers = [proposer.address];
+    const executors = [proposer.address];
+
+    const TimelockC = await ethers.getContractFactory("TimelockController");
+    const minDelay = 1;
+    timelock = await TimelockC.deploy(minDelay, proposers, executors, owner.address);
+    await timelock.waitForDeployment();
+
+    tokenAddr = await token.getAddress();
+    token2Addr = await token2.getAddress();
+
+    const Governance = await ethers.getContractFactory("ZDAO");
+
+    governance = await Governance.deploy(
+      "Governance DAO",
+      tokenAddr,
+      await timelock.getAddress(),
+      1n,
+      5n,
+      1n,
+      0n,
+      2n
+    );
+    await governance.waitForDeployment();
+
+    await timelock.connect(owner).grantRole(await timelock.PROPOSER_ROLE(), await governance.getAddress());
+    await timelock.connect(owner).grantRole(await timelock.EXECUTOR_ROLE(), await governance.getAddress());
+
+    await token.connect(owner).grantRole(MINTER_ROLE, await timelock.getAddress());
+    await token2.connect(owner).grantRole(MINTER_ROLE, await timelock.getAddress());
+  });
+
+  describe("ERC20 Voting Tests", () => {
+    it("Many users vote (+ and -)", async () => {
+      const mintAmount = ethers.parseUnits("100");
+      const targets = [tokenAddr];
+      const values = [0];
+      const calldatas = [
+        token.interface.encodeFunctionData("mint", [proposer.address, mintAmount]),
+      ];
+      const description = "Mint additional tokens to proposer";
+
+      await governance.connect(proposer).propose(targets, values, calldatas, description);
+
+      const descriptionHash = ethers.keccak256(ethers.toUtf8Bytes(description));
+      const proposalId = await governance.hashProposal(targets, values, calldatas, descriptionHash);
+
+      await mineBlocks(2);
+
+      await governance.connect(proposer).castVote(proposalId, 1);
+      await governance.connect(voter1).castVote(proposalId, 1);
+      await governance.connect(voter2).castVote(proposalId, 0);
+
+      await mineBlocks(5);
+
+      const state = await governance.state(proposalId);
+      expect(state).to.equal(4);
+
+      await governance.connect(proposer).queue(
+        targets,
+        values,
+        calldatas,
+        descriptionHash
+      );
+
+      await ethers.provider.send("evm_increaseTime", [3600]);
+      await ethers.provider.send("evm_mine", []);
+      await governance.connect(proposer).execute(targets, values, calldatas, descriptionHash);
+
+      const finalBalance = await token.balanceOf(proposer.address);
+      // 1000 was before mint and then mints 100 more.
+      expect(finalBalance).to.equal(ethers.parseUnits("1100"));
```

Comment:
why is this not a variable that we get from the code? some balanceBefore + mintAmt or something